Layers/Renderers/Objects/ObjectMultiMesh.gd

```diff
@@ -25,47 +32,67 @@ func rebuild_aabb(node):
 
 
 func override_build(center_x, center_y):
-	fresh_multimesh = MultiMesh.new()
-	
-	fresh_multimesh.transform_format = MultiMesh.TRANSFORM_3D
-	fresh_multimesh.mesh = object
+	fresh_multimesh_mapping = {}
 	
 	var top_left_x = float(center_x - size / 2)
 	var top_left_y = float(center_y + size / 2)
 	
 	features = object_layer.get_features_in_square(top_left_x, top_left_y, size, 10000000)
-	fresh_multimesh.instance_count = features.size()
 	
-	rng.state = initial_rng_state
-
-	var i = 0
-	for feature in features:
-		var instance_scale = randf_range(0.9, 1.2) if randomize else 1.0
-		var instance_rotation = float(feature.get_attribute("LL_rot"))
+	# Function to filter objects depending on their attribute type
+	var filter = func(f, key): 
+		# Check for defaults
+		if  selector_attribute_name == "": 
+			return true
+		var type = f.get_attribute(selector_attribute_name)
+		if type not in objects_mapping.keys() and key == "default":
+			return true
 		
-		var pos = feature.get_offset_vector3(-int(center_x), 0, -int(center_y))
-		pos.y = height_layer.get_value_at_position(pos.x + center_x, center_y - pos.z)
+		# Check for right type
+		return type == key
 		
-		var new_transform = Transform3D().scaled(Vector3(instance_scale, instance_scale, instance_scale))
+	for key in objects_mapping:
+		var filtered = features.filter(filter.bind(key))
+		var object = load(objects_mapping[key])
```

Comment:
This might impact performance a lot, would be better to create a key-to-instance mapping and `duplicate` those right?